### PR TITLE
Add randomization in sync retry batch peer selection

### PR DESCRIPTION
**Pull request by dapplion**
## Issue Addressed

When retrying a batch, peers are prioritized by:
1. They have some failures
2. Count of active requets
3. Break tie with NodeID value

While the prioritization is okay, breaking ties by NodeID is not fair and prioritizes peers that just happen to have a peer ID if a smaller value or those that have farmed peer IDs with smaller values.

## Proposed Changes

Add a random u32 before sorting batch retry peers

### CODE REVIEW

File: beacon_node/network/src/sync/backfill_sync/mod.rs [hunk 1]
```diff
@@ -900,13 +900,14 @@ impl<T: BeaconChainTypes> BackFillSync<T> {
...
                         self.active_requests.get(peer).map(|v| v.len()).unwrap_or(0),
+                        rand::thread_rng().gen::<u32>(),
                         *peer,
                     )
                 })
                 .collect::<Vec<_>>();
             // Sort peers prioritizing unrelated peers with less active requests.
             priorized_peers.sort_unstable();
-            priorized_peers.first().map(|&(_, _, peer)| peer)
+            priorized_peers.first().map(|&(_, _, _, peer)| peer)
```
Comment (jxs) [inline beacon_node/network/src/sync/backfill_sync/mod.rs]: not related with this PR, but do we need to iterate, collect, and then sort? To get the lowest peer by that `Ord` criteria we can just iterate once and get the lowest with :
```rust
let new_peer = self
            .network_globals
            .peers
            .read()
            .synced_peers()
            .map(|peer| {
                (
                    failed_peers.contains(peer),
                    self.active_requests.get(peer).map(|v| v.len()).unwrap_or(0),
                    rand::thread_rng().gen::<u32>(),
                    *peer,
                )
            })
            .min()
            .map(|(_, _, _, peer)| peer);
```

same applies to the other change on this PR

**Review by jxs — COMMENTED** [review]
the rationale for the change makes sense to me! Left a comment wrt to the implementation only

File: beacon_node/network/src/sync/backfill_sync/mod.rs (hunk 1 above)
Comment (dapplion) [reply beacon_node/network/src/sync/backfill_sync/mod.rs]: Nice suggestion! Pushed

File: beacon_node/network/src/sync/backfill_sync/mod.rs [hunk 2]
```diff
@@ -19,7 +19,7 @@ use beacon_chain::block_verification_types::RpcBlock;
 use beacon_chain::{BeaconChain, BeaconChainTypes};
 use lighthouse_network::types::{BackFillState, NetworkGlobals};
 use lighthouse_network::{PeerAction, PeerId};
-use rand::seq::SliceRandom;
+use rand::{seq::SliceRandom, Rng};
```
Comment (jxs) [inline beacon_node/network/src/sync/backfill_sync/mod.rs]:
```suggestion
use rand::seq::SliceRandom;
```

File: beacon_node/network/src/sync/backfill_sync/mod.rs [hunk 3]
```diff
@@ -890,24 +890,22 @@ impl<T: BeaconChainTypes> BackFillSync<T> {
...
+        let new_peer = self
+            .network_globals
+            .peers
+            .read()
+            .synced_peers()
+            .map(|peer| {
+                (
+                    failed_peers.contains(peer),
+                    self.active_requests.get(peer).map(|v| v.len()).unwrap_or(0),
+                    rand::thread_rng().gen::<u32>(),
```
Comment (jxs) [inline beacon_node/network/src/sync/backfill_sync/mod.rs]: nit:
```suggestion
                    rand::random::<u32>(),
```
and we can avoid importing `Rng` above.

**Review by jxs — APPROVED** [review]
LGTM lion! Left one final suggestion which can just be applied

**Comment by jxs** [comment]
@mergifyio queue

**Comment by jxs** [comment]
@mergify refresh

**Review by AgeManning — APPROVED** [review]

**Comment by jxs** [comment]
@mergifyio queue

**Pull request merged**